pkgs/development/compilers/llvm/common/flang/default.nix

```diff
@@ -0,0 +1,69 @@
+{
+  stdenv,
+  llvm_meta,
+  buildLlvmTools,
+  monorepoSrc,
+  runCommand,
+  cmake,
+  libclang,
+  libllvm,
+  libxml2,
+  lit,
+  mlir,
+  ninja,
+  version,
+}:
+stdenv.mkDerivation rec {
+  pname = "flang";
+  inherit version;
+
+  src = runCommand "${pname}-src-${version}" { } ''
+    mkdir -p "$out"
+    cp -r ${monorepoSrc}/cmake "$out"/cmake
+    cp -r ${monorepoSrc}/flang-rt "$out"/flang-rt
+    cp -r ${monorepoSrc}/flang "$out"/flang
+  '';
+
+  sourceRoot = "${src.name}/${pname}";
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+  ];
+  buildInputs = [
+    libclang
+    libllvm
+    libxml2
+    mlir
+  ];
+
+  cmakeFlags = [
+    "-DCLANG_DIR=${libclang.dev}/lib/cmake/clang"
+    "-DMLIR_DIR=${mlir.dev}/lib/cmake/mlir"
+    "-DLLVM_DIR=${libllvm.dev}/lib/cmake/llvm"
+    "-DMLIR_TABLEGEN_EXE=${buildLlvmTools.tblgen}/bin/mlir-tblgen"
+    "-DLLVM_EXTERNAL_LIT=${lit}/bin/lit"
+    "-DLLVM_LIT_ARGS=-v"
+    # Tests depend on test libs from MLIR which are only available in a
+    # build tree.
+    "-DFLANG_INCLUDE_TESTS=OFF"
+  ];
+
+  passthru = {
+    # Used by cc-wrapper to determine whether or not the default setup hook is enabled.
+    langC = false;
+    langCC = false;
+    langFortran = true;
+    isClang = true;
+
+    hardeningUnsupportedFlags = [
+      "zerocallusedregs"
+      "stackprotector"
+    ];
```

Comment:
Why aren't these on the toplevel and inside `passthru`?

Reply:
Hmmm not sure I follow. They're inside passthru here and get picked up by cc-wrapper, right? IIRC I looked at what clang/gcc did here.